applications/IgaApplication/custom_elements/shell_5p_hierarchic_element.h

```diff
@@ -0,0 +1,613 @@
+        const Vector& rw,
+        const Vector& rDw_D1,
+        const Vector& rDw_D2,
+        const Vector& ra1,
+        const Vector& ra2);
+
+    /// @brief This method performs a transformation of a curvilinear strain vector with Voigt size 5 to a Cartesian strain vector with Voigt size 6
+    void TransformationCurvilinearStrainSize5ToCartesianStrainSize6(
+        const Vector& rCurvilinearStrain,
+        Vector& rCartesianStrain);
+
+    /* @brief The method calculates the B matrix of the Kirchhoff-Love contributions
+     * @param rB = B matrix
+     */
+    void CalculateB(
+        Matrix& rB,
+        const MetricVariables& rActualMetric,
+        IndexType IntegrationPointIndex = 0);
+
+    /// @brief The method calculates the second variations according to Kirchhoff-Love
+    void CalculateSecondVariations(
+        SecondVariations& rSecondVariations,
+        const MetricVariables& rActualMetric,
+        IndexType IntegrationPointIndex = 0);
+
+    /* @brief The method calculates the additional terms of the B matrix and the second variations according to Reissner-Mindlin
+     * @param rw = hierarchic shear difference vector
+     * @param rDw_D1 = derivative of the shear difference vector w.r.t. theta1
+     * @param rDw_D2 = derivative of the shear difference vector w.r.t. theta2
+     * @param rw_alpha = components of the shear difference vector
+     * @param rDw_alpha_Dbeta = derivative of the component alpha of the shear difference vector w.r.t. theta(beta) which means theta1 or theta2
+     */
+    void CalculateVariationsReissnerMindlin(
+        Matrix& rB,
+        SecondVariations& rSecondVariations,
+        const Vector& rw,
+        const Vector& rDw_D1,
+        const Vector& rDw_D2,
+        const Vector& rw_alpha,
+        const Matrix& rDw_alpha_Dbeta,
+        const MetricVariables& rActualMetric,
+        const bool& rCalculateStiffnessMatrixFlag,
+        IndexType IntegrationPointIndex = 0);
+ 
+    /// @brief Stress recovery
+    void Calculate(
+        const Variable<double>& rVariable,
+        double& rValues,
+        const ProcessInfo& rCurrentProcessInfo) override;
+
+    ///@}
+
+private:
+    ///@name Private member Variables
+    ///@{
+
+    /// The vector containing the constitutive laws
+    std::vector<ConstitutiveLaw::Pointer> mConstitutiveLawVector;
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+    /// @brief It initializes the material
+    virtual void InitializeMaterial();
+
+    /// Computes the Hessian of this problem
+    void CalculateHessian(
+        Matrix& Hessian,
+        const Matrix& rDDN_DDe);
+
+    ///@name Serialization
+    ///@{
+
+    friend class Serializer;
+
+    void save(Serializer& rSerializer) const override
+    {
+        KRATOS_SERIALIZE_SAVE_BASE_CLASS(rSerializer, Element)
+    }
+
+    void load(Serializer& rSerializer) override
+    {
+        KRATOS_SERIALIZE_LOAD_BASE_CLASS(rSerializer, Element)
+    }

```

Comment:
missing implementation

Reply:
Hehe, it's actually not that straight away, to be honest, I'd prefer to keep this for future work. I'll soon add up the internal stresses and stuff and there'll be a reconsideration anyways. Thusly, i prefer to postpone this. In iga we're never using that anyways...

Reply:
> Thusly, i prefer to postpone this

ok for me but I think this is still important

Not having this means you cannot restart which makes IGA basically useless for Clusters where you have to restart due to runtime simulations
Also some real life cases of IGA with embedded FSI will have to be done on the cluster so I would add it soon!